e2e_tests/resources/workspace.py

```diff
@@ -41,24 +39,13 @@ async def get_workspace_auth_details(admin_token, workspace_id, verify) -> Tuple
     async with AsyncClient(verify=verify) as client:
         auth_headers = get_auth_header(admin_token)
         scope_uri = await get_identifier_uri(client, workspace_id, auth_headers)
+    if config.TEST_ACCOUNT_CLIENT_ID != "" and config.TEST_ACCOUNT_CLIENT_SECRET != "":
+        # Logging in as an Enterprise Application: Use Client Credentials flow
+        credential = ClientSecretCredential(config.AAD_TENANT_ID, config.TEST_ACCOUNT_CLIENT_ID, config.TEST_ACCOUNT_CLIENT_SECRET, connection_verify=verify, authority=cloud.get_authority_domain())
+        token = credential.get_token(f'{scope_uri}/.default')
+    else:
+        # Logging in as a User: Use Resource Owner Password Credentials flow
+        credential = UsernamePasswordCredential(config.TEST_APP_ID, config.TEST_USER_NAME, config.TEST_USER_PASSWORD, connection_verify=verify, authority=cloud.get_authority_domain(), tenant_id=config.AAD_TENANT_ID)
+        token = credential.get_token(f'{scope_uri}/user_impersonation')
 
-        if config.TEST_ACCOUNT_CLIENT_ID != "" and config.TEST_ACCOUNT_CLIENT_SECRET != "":
-            # Logging in as an Enterprise Application: Use Client Credentials flow
-            payload = f"grant_type=client_credentials&client_id={config.TEST_ACCOUNT_CLIENT_ID}&client_secret={config.TEST_ACCOUNT_CLIENT_SECRET}&scope={scope_uri}/.default"
-            url = f"{AAD_AUTHORITY_URL}/{config.AAD_TENANT_ID}/oauth2/v2.0/token"
-
-        else:
-            # Logging in as a User: Use Resource Owner Password Credentials flow
-            payload = f"grant_type=password&resource={workspace_id}&username={config.TEST_USER_NAME}&password={config.TEST_USER_PASSWORD}&scope={scope_uri}/user_impersonation&client_id={config.TEST_APP_ID}"
-            url = f"{AAD_AUTHORITY_URL}/{config.AAD_TENANT_ID}/oauth2/token"
-
-        response = await client.post(url, headers=auth_headers, content=payload)
-        try:
-            responseJson = response.json()
-        except JSONDecodeError:
-            raise Exception("Failed to parse response as JSON: {}".format(response.content))
-
-        if "access_token" not in responseJson or response.status_code != status.HTTP_200_OK:
-            raise Exception("Failed to get access_token: {}".format(response.content))
-
-        return responseJson["access_token"], scope_uri
+    return token.token, scope_uri
```

Comment:
This looks to be very similar to the block above. Is it possible to extract into a generic method like _get_token_ in the helpers class?

Reply:
Done